src/SSVOpenHexagon/Core/HexagonGame.cpp

```diff
@@ -243,19 +251,25 @@ void HexagonGame::checkAndSaveScore()
     }
     else
     {
-        if(status.scoreInvalid || !Config::isEligibleForScore())
-        {
-            ssvu::lo("hg::HexagonGame::checkAndSaveScore()")
-                << "Not sending/saving score - not eligible\n"
-                << Config::getUneligibilityReason() << "\n";
-            return;
-        }
+        // These are requirements that need to be met for a score to be sent online
         if(time < 8)
         {
             ssvu::lo("hg::HexagonGame::checkAndSaveScore()")
                 << "Not sending score - less than 8 seconds\n";
             return;
         }
+		if(Online::getServerVersion() == -1)
+		{
+			ssvu::lo("hg::HexagonGame::checkAndSaveScore()")
+                << "Not sending score - connection error\n";
+			return;
+		}
+		if(Online::getServerVersion() > Config::getVersion())
+		{
+			ssvu::lo("hg::HexagonGame::checkAndSaveScore()")
+                << "Not sending score - version mismatch\n";
+			return;
+		}
```

Comment:
I'm not sure why you put these here and removed the check on `if(status.scoreInvalid || !Config::isEligibleForScore())` - can you explain?

Reply:
Since I have now moved isEligibleForScore to be called before it checks whether the user is local or not, I had to remove the online checks from it, otherwise the score would otherwise be immediately disqualified; it can't connect to the server because there's no server to connect to.

These checks should only be happening when we know that the player is trying to submit a score online and they aren't a local profile, hence I removed them completely from isEligibleForScore and put them in the online score checking session, which is where they belong. 

And the original if statement is not gone. I decided to split the conditions into two separate if statements because you can't get an uneligibility reason if you manually invalidate the score. Hence a separate error message had to be made.